elasticsearch/src/main/java/org/hibernate/search/elasticsearch/settings/impl/SimpleAnalysisDefinitionFactory.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.elasticsearch.settings.impl;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.hibernate.search.annotations.Parameter;
+import org.hibernate.search.elasticsearch.settings.impl.model.AnalysisDefinition;
+import org.hibernate.search.exception.AssertionFailure;
+
+import com.google.gson.JsonElement;
+
+class SimpleAnalysisDefinitionFactory<D extends AnalysisDefinition> implements AnalysisDefinitionFactory<D> {
+
+	private final Class<D> targetClass;
+	private final String type;
+	private final Map<String, String> parameterNameTranslations;
+	private final Map<String, ParameterValueTransformer> parameterValueTranslations;
+	private final Map<String, JsonElement> staticParameters;
+
+	public SimpleAnalysisDefinitionFactory(Class<D> targetClazz, String type, Map<String, String> parameterNameTranslations,
+			Map<String, ParameterValueTransformer> parameterValueTranslations,
+			Map<String, JsonElement> staticParameters) {
+		super();
```

Comment:
Mmmh, OK, any reason why you add a `super();` everywhere? I can see a pattern here :).